activerecord/lib/active_record/reflection.rb

```diff
@@ -1038,5 +1039,7 @@ def aliased_table
 
       def all_includes; yield; end
     end
+
+    ActiveSupport.run_load_hooks(:active_record_reflection, self)
```

Comment:
I think we can do this without adding a new load hook to the public API. The `class_attribute` should be added to the `ActiveRecord::Base` class. The `ReflectionExtensions` can be added directly to `ActiveRecord::Reflection` without need a `on_load` hook.

Reply:
Sure that makes sense. I was just trying to keep the ActiveStorage stuff to itself, but happy to move it over.